os/fs/driver/mtd/smart.c

```diff
@@ -4654,107 +4426,12 @@ static int smart_writesector(FAR struct smart_struct_s *dev, unsigned long arg)
 		ret = smart_bytewrite(dev, offset, req->count, req->buffer);
 		if (ret != req->count) {
 			fdbg("Error committing physical sector %d\n", physsector);
-			ret = -EIO;
-			goto errout;
-		}
-		/* check BAD Sector or not. */
-
-#ifdef CONFIG_SMARTFS_BAD_SECTOR
-
-		int bsm_ret = MTD_BREAD(dev->mtd, physsector * dev->mtdBlksPerSector,
-								dev->mtdBlksPerSector, (FAR uint8_t *)dev->bad_sector_rwbuffer);
-		if (bsm_ret < 0) {
-			fdbg("read %d phy sector data for BSM failed, bsm_ret = %d\n", physsector, bsm_ret);
-		}
-		int i;
-
-		for (i = 0; i < req->count; i++) {
-			uint8_t read_byte = dev->bad_sector_rwbuffer[sizeof(struct smart_sect_header_s)
-								+ req->offset + i];
-			if (req->buffer[i] != read_byte) {
-				bad_sector_found = true;
-				fwdbg("Sorry! %d sector is bad.\n", physsector);
-				break;
-			}
+			return -EIO;
 		}
-#endif							/* END OF CONFIG_SMARTFS_BAD_SECTOR */
 #endif
 	}
 
 	ret = OK;
-
-errout:
-#ifdef CONFIG_SMARTFS_BAD_SECTOR
-
-	if (bad_sector_found) {
-		needsrelocate = TRUE;
-		dev->badSectorList[physsector] = TRUE;
-
-#if CONFIG_SMARTFS_ERASEDSTATE == 0xFF
-		byte = header->status & ~(SMART_STATUS_RELEASED | SMART_STATUS_COMMITTED);
-#else
-		byte = header->status | SMART_STATUS_RELEASED | SMART_STATUS_COMMITTED;
-#endif
-		offset = physsector * dev->geo.blocksize + offsetof(struct smart_sect_header_s, status);
-		ret = smart_bytewrite(dev, offset, 1, &byte);
-		if (ret != 1) {
-			fdbg("Error committing physical sector %d\n", physsector);
-			ret = -EIO;
-			goto errout;
-		}
-		/* Update releasecount for bad sector and freecount for the
-		 * newly allocated but bad physical sector. */
-
-		block = physsector / dev->sectorsPerBlk;
-#ifdef CONFIG_MTD_SMART_PACK_COUNTS
-		smart_add_count(dev, dev->releasecount, block, 1);
-		smart_add_count(dev, dev->freecount, physsector / dev->sectorsPerBlk, -1);
-#else
-		dev->releasecount[block]++;
```

Comment:
same for release counts and other such variables ?

Reply:
As I wrote above, it released so count should be added.